packages/swap-manager/src/implementation/oneinch/OneInchSwapManager.ts

```diff
@@ -0,0 +1,102 @@
+import { Address, Percentage, TokenAmount } from '@summerfi/sdk/common'
+import { SwapManager } from '~swap-manager'
+import { SwapData, SwapProvider } from '~swap-manager/interfaces/SwapManager'
+import { AuthHeader, AuthHeaderKey } from './types'
+import { ChainInfo } from '@summerfi/sdk/chains'
+import axios from 'axios'
+
+export class OneInchSwapManager implements SwapManager {
+  private readonly _OneInchApiURL = 'https://api-oasis.1inch.io'
+  private readonly _OneInchVersion = 'v4.0'
+  private readonly _AllowedSwapProtocols = [
+    'UNISWAP_V3',
+    'PMM1',
+    'PMM2',
+    'PMM3',
+    'PMM4',
+    'UNISWAP_V2',
+    'SUSHI',
+    'CURVE',
+    'CURVE_V2',
+    'PSM',
+    'WSTETH',
+    'BALANCER',
+    'BALANCER_V2',
+    'BALANCER_V2_WRAPPER',
+    'ST_ETH',
+    'WETH',
+    'ROCKET_POOL',
+  ]
+
+  private readonly _apiKey: string
+
+  constructor() {
+    if (!process.env.ONE_INCH_API_KEY) {
```

Comment:
All parameters should be passed through the constructor. Easier to test

Reply:
We should pass the allowed protocols and 1inch version. 
The names for protocols vary across networks.
I recommend to have an interface `IOneInchClient` (it's a client, not manager) and have a different implementation based on the network.